### ISSUE TITLE
Unicode Error

### ISSUE THREAD

**Issue by akeilox** [rod0/thermomin#3]
I have followed the installation instructions and I have openCV and numpy running for other applications, so I switched to same environment before testing thermomin.

After installing the service OK, and running simpletest.py I am getting this:

Traceback (most recent call last):
  File "simpletest.py", line 8, in <module>
    ir = numpy.frombuffer(fifo.read()[0:256], dtype=numpy.float32)
  File "/home/pi/.virtualenvs/cv/lib/python3.5/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 1: invalid continuation byte

Can someone shed some light on what might be the issue ? The virtualenv opencv and numpy working fine for other apps, so not sure what is likely the issue here.

**Comment by prabrgit** [comment]
I am also getting same error. Any update?

**Comment by rlowit** [comment]
I dont have the project alive, but can you change:
ir = numpy.frombuffer(fifo.read()[0:256], dtype=numpy.float32) 
to:
ir = numpy.frombuffer(fifo.read(256), dtype=numpy.float32)
and see if it fix the problem?
By changing fifo.read()[0:256] to fifo.read(256), you're simply reading 256 bytes of the file instead of all the bytes it contains. This is useful if you only need to read a specific amount of bytes from the file.

In other words, the numpy.frombuffer() function will interpret the 256 bytes read as a sequence of numpy.float32 floating-point values, without trying to decode them in any way. Therefore, if the data is binary, it will simply be read and interpreted as floating-point numbers, without any need for encoding.
